server/sonar-process/src/main/java/org/sonar/process/jmx/JmxConnectionFactory.java

```diff
@@ -44,14 +45,23 @@ public JmxConnectionFactory(Props props) {
     this.ipcSharedDir = props.nonNullValueAsFile(ProcessEntryPoint.PROPERTY_SHARED_PATH);
   }
 
+  @CheckForNull
   public JmxConnection create(ProcessId processId) {
     try (DefaultProcessCommands commands = DefaultProcessCommands.secondary(ipcSharedDir, processId.getIpcIndex())) {
-      String url = commands.getJmxUrl();
-      JMXConnector jmxConnector = JMXConnectorFactory.newJMXConnector(new JMXServiceURL(url), null);
-      jmxConnector.connect();
-      return new JmxConnection(jmxConnector);
+      if (commands.isUp()) {
+        String url = commands.getJmxUrl();
+        JMXConnector jmxConnector = JMXConnectorFactory.newJMXConnector(new JMXServiceURL(url), null);
+        jmxConnector.connect();
+        return new JmxConnection(jmxConnector);
+      }
+      return null;
     } catch (Exception e) {
       throw new IllegalStateException("Can not connect to process " + processId, e);
     }
   }
+
+  // visible for testing
+  File getIpcSharedDir() {
```

Comment:
No, the constructor with `File` is used by `JmxConnectionFactoryProvider` which does not have `Props`.